src/main/java/io/confluent/connect/jdbc/source/TimestampIncrementingTableQuerier.java

```diff
@@ -171,9 +174,17 @@ public SourceRecord extractRecord() throws SQLException {
     final Map<String, String> partition;
     switch (mode) {
       case TABLE:
-        String name = tableId.tableName(); // backward compatible
-        partition = Collections.singletonMap(JdbcSourceConnectorConstants.TABLE_NAME_KEY, name);
-        topic = topicPrefix + name;
+        if (useFqn) {
+          String fqn = ExpressionBuilder.create().append(tableId, false).toString();
+          partition = new HashMap<>();
+          partition.put(JdbcSourceConnectorConstants.TABLE_NAME_KEY, fqn);
+          partition.put(JdbcSourceConnectorConstants.OFFSET_PROTOCOL_VERSION, "1");
+          topic = topicPrefix + fqn;
+        } else {
+          String name = tableId.tableName(); // backward compatible
+          partition = Collections.singletonMap(JdbcSourceConnectorConstants.TABLE_NAME_KEY, name);
+          topic = topicPrefix + name;
+        }
```

Comment:
Seems like all of this could be extracted into a strategy function to compute the topic given the table ID (e.g., `topicName(TableId)`), and a second method to compute the partitions for a table ID (e.g., `sourcePartitions(TableId)`). We could have separate strategy implementations for each protocol version, and a particular strategy implementation can be passed into this class.